tests/test_solve_saveat.py

```diff
@@ -0,0 +1,136 @@
+"""Tests for solve_saveat()."""
+import jax
+import jax.numpy as jnp
+import pytest_cases
+import scipy.integrate
+
+from tornadox import ivp_examples, solve
+from tornadox.blocks import markov
+from tornadox.solvers import ek1
+
+
+def case_ivp_vanderpol_nonstiff():
+    f, tspan, u0 = ivp_examples.vanderpol(stiffness_constant=0.1)
+    return f, tspan, u0
+
+
+@pytest_cases.parametrize("num_derivatives", [3, 6])
+def case_solver_ek1(num_derivatives):
+
+    # Assume that the test-IVPs are two-dimensional.
+    # When this changes, update here.
+    ode_dimension = 2
+
+    return ek1.ek1_saveat(ode_dimension=ode_dimension, num_derivatives=num_derivatives)
+
+
+@pytest_cases.parametrize_with_cases(argnames=("ivp",), cases=".", prefix="case_ivp_")
+@pytest_cases.parametrize_with_cases(
+    argnames=("solver",), cases=".", prefix="case_solver_"
+)
+@pytest_cases.parametrize("atol, rtol", ((1e-6, 1e-3),))
+def test_solve_ivp_saveat(ivp, solver, atol, rtol):
+    f, tspan, u0 = ivp
+    df = jax.jacfwd(f)
+    f, df = jax.jit(f), jax.jit(df)
+
+    t0, t1 = tspan
+    saveat = jnp.array([t0, 0.75 * (t0 + t1), t1])
+
+    solution = solve.solve_ivp_saveat(
+        f=f, df=df, saveat=saveat, u0=u0, solver=solver, atol=atol, rtol=rtol
+    )
+    # Each of those is of the form ((rv, backward_model), t)
+    state_terminal, solution_stacked, state_initial = solution
+
+    # Uhmmm, each output (terminal, stacked, initial) looks so innocent,
+    # but they are _packed_ full of information.
+    # Long list of assertions incoming...
+    # This probably implies that the output needs some refactoring,
+    # but for now I am happy with it, because it does the job.
```

Comment:
it might be more usable if we stack this into a suitable namedtuple, for example. But so far, I like to keep it so naked, and I'd say we can iterate if it bothers anyone too much?